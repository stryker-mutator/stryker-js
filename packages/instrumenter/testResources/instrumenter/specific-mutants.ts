const a = 1 + 1;
const b = 1 - 1;

if (a === 2 && b === 0) {
  console.log('a');
}

if (a === 2 && b === 0) {
  console.log('b');
}

const itemWithLongName = {
  longPropertyName1: 1,
  longPropertyName2: 2,
  longPropertyName3: 3,
};

const item = () =>
  itemWithLongName.longPropertyName1 === itemWithLongName.longPropertyName2 &&
  itemWithLongName.longPropertyName1 === itemWithLongName.longPropertyName3;
