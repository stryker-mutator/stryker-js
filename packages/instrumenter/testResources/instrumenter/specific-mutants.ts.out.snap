// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instrumenter integration with mutation ranges should only mutate specific mutants for the given file 1`] = `
"function stryNS_9fa48() {
  var g = new Function(\\"return this\\")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

const a = stryMutAct_9fa48(\\"0\\") ? 1 - 1 : (stryCov_9fa48(\\"0\\"), 1 + 1);
const b = 1 - 1;

if ((stryMutAct_9fa48(\\"2\\") ? a !== 2 : stryMutAct_9fa48(\\"1\\") ? true : (stryCov_9fa48(\\"1\\", \\"2\\"), a === 2)) && b === 0) {
  console.log('a');
}

if (a === 2 && (stryMutAct_9fa48(\\"4\\") ? b !== 0 : stryMutAct_9fa48(\\"3\\") ? true : (stryCov_9fa48(\\"3\\", \\"4\\"), b === 0))) {
  console.log('b');
}

const itemWithLongName = {
  longPropertyName1: 1,
  longPropertyName2: 2,
  longPropertyName3: 3
};

const item = () => stryMutAct_9fa48(\\"7\\") ? itemWithLongName.longPropertyName1 === itemWithLongName.longPropertyName2 || itemWithLongName.longPropertyName1 === itemWithLongName.longPropertyName3 : stryMutAct_9fa48(\\"6\\") ? false : stryMutAct_9fa48(\\"5\\") ? true : (stryCov_9fa48(\\"5\\", \\"6\\", \\"7\\"), (stryMutAct_9fa48(\\"9\\") ? itemWithLongName.longPropertyName1 !== itemWithLongName.longPropertyName2 : stryMutAct_9fa48(\\"8\\") ? true : (stryCov_9fa48(\\"8\\", \\"9\\"), itemWithLongName.longPropertyName1 === itemWithLongName.longPropertyName2)) && (stryMutAct_9fa48(\\"11\\") ? itemWithLongName.longPropertyName1 !== itemWithLongName.longPropertyName3 : stryMutAct_9fa48(\\"10\\") ? true : (stryCov_9fa48(\\"10\\", \\"11\\"), itemWithLongName.longPropertyName1 === itemWithLongName.longPropertyName3)));"
`;
